### PR TITLE
fixes toc file buffer

### PR DESCRIPTION
**Pull request by eginez**
Only grow the toc file buffer if we run out of space

### CODE REVIEW

File: content-store/src/main/java/nl/inl/blacklab/contentstore/ContentStoreFixedBlockWriter.java [hunk 1]
```diff
@@ -203,22 +203,23 @@ public void clear() throws IOException {
     }
 
     @Override
-    protected void mapToc(boolean writeable) throws IOException {
+    protected void mapToc(boolean writeable, boolean growBuffer) throws IOException {
         tocRaf = new RandomAccessFile(tocFile, writeable ? "rw" : "r");
         long fl = tocFile.length();
-        if (writeable) {
```
Comment (brendan-lexion) [inline content-store/src/main/java/nl/inl/blacklab/contentstore/ContentStoreFixedBlockWriter.java:209]: Is it really safe to ignore writeable here? Maybe the condition should be `writeable && (growBuffer || fl == 0)`?

Comment (eginez) [reply content-store/src/main/java/nl/inl/blacklab/contentstore/ContentStoreFixedBlockWriter.java:209]: it actually does not matter. writable=`true` only from the place that initiated the leak and when we are running out of space. We only need to grow that buffer in one of those cases, but checking for growth while reading makes sense

Comment (brendan-lexion) [reply content-store/src/main/java/nl/inl/blacklab/contentstore/ContentStoreFixedBlockWriter.java:209]: Gotcha. Appreciate the extra safety check!

**Review by brendan-lexion — APPROVED** [review]
LGTM!

**Pull request merged**